storm-core/src/jvm/org/apache/storm/daemon/StormCommon.java

```diff
@@ -0,0 +1,605 @@
+import org.apache.storm.metric.EventLoggerBolt;
+import org.apache.storm.metric.MetricsConsumerBolt;
+import org.apache.storm.metric.SystemBolt;
+import org.apache.storm.security.auth.IAuthorizer;
+import org.apache.storm.task.IBolt;
+import org.apache.storm.testing.NonRichBoltTracker;
+import org.apache.storm.utils.ConfigUtils;
+import org.apache.storm.utils.IPredicate;
+import org.apache.storm.utils.ThriftTopologyUtils;
+import org.apache.storm.utils.Utils;
+import org.json.simple.JSONValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+public class StormCommon {
+    // A singleton instance allows us to mock delegated static methods in our
+    // tests by subclassing.
+    private static StormCommon _instance = new StormCommon();
+
+    /**
+     * Provide an instance of this class for delegates to use.  To mock out
+     * delegated methods, provide an instance of a subclass that overrides the
+     * implementation of the delegated method.
+     * @param common a StormCommon instance
+     * @return the previously set instance
+     */
+    public static StormCommon setInstance(StormCommon common) {
+        StormCommon oldInstance = _instance;
+        _instance = common;
+        return oldInstance;
+    }
+
+    private static final Logger LOG = LoggerFactory.getLogger(StormCommon.class);
+
+    public static final String ACKER_COMPONENT_ID = Acker.ACKER_COMPONENT_ID;
+    public static final String ACKER_INIT_STREAM_ID = Acker.ACKER_INIT_STREAM_ID;
+    public static final String ACKER_ACK_STREAM_ID = Acker.ACKER_ACK_STREAM_ID;
+    public static final String ACKER_FAIL_STREAM_ID = Acker.ACKER_FAIL_STREAM_ID;
+
+    public static final String SYSTEM_STREAM_ID = "__system";
+
+    public static final String EVENTLOGGER_COMPONENT_ID = "__eventlogger";
+    public static final String EVENTLOGGER_STREAM_ID = "__eventlog";
+
+    public static void startMetricsReporter(PreparableReporter report, Map conf) {
+        report.prepare(new MetricRegistry(), conf);
+        report.start();
+        LOG.info("Started statistics report plugin...");
+    }
+
+    public static void startMetricsReporters(Map conf) {
+        List<PreparableReporter> reporters = MetricsUtils.getPreparableReporters(conf);
+        for (PreparableReporter reporter : reporters) {
+            startMetricsReporter(reporter, conf);
+        }
+    }
+
+    public static String getTopologyNameById(String topologyId) {
+        String topologyName = null;
+        try {
+            topologyName = topologyIdToName(topologyId);
+        } catch (InvalidTopologyException e) {
+            LOG.error("Invalid topologyId=" + topologyId);
+        }
+        return topologyName;
+    }
+
+    /**
+     * Convert topologyId to topologyName. TopologyId = topoloygName-counter-timeStamp
+     *
+     * @param topologyId
+     * @return
+     */
+    public static String topologyIdToName(String topologyId) throws InvalidTopologyException {
```

Comment:
In fact the above function can be removed since an invalid name is never expected.